utils/cache/backend_redis.c

```diff
@@ -91,6 +93,34 @@ static status_t redis_set(redisContext* c, const char* const key, const char* co
   return ret;
 }
 
+int redis_db_used_size(redisContext* c) {
+  status_t ret = SC_OK;
+  const char size_field[] = "used_memory:";
+  redisReply* reply = redisCommand(c, "INFO");
+  if (reply->type == REDIS_REPLY_STRING) {
+    // Parsing the result returned by redis. Get informationa from field `used_memory`
+    char* str = strstr(reply->str, size_field);
+    char* size_str = strtok(str, "\n");
+    char* ptr = size_str + strlen(size_field);
+
+    char* strtol_p = NULL;
+    long int strtol_temp;
+    strtol_temp = strtol(ptr, &strtol_p, 10);
+    if (strtol_p != ptr && errno != ERANGE && strtol_temp >= INT_MIN && strtol_temp <= INT_MAX) {
+      freeReplyObject(reply);
+      return (int)strtol_temp;
+    } else {
+      ta_log_error("Malformed input or illegal input character\n");
```

Comment:
Besides error log, should we throw exception code?

Reply:
If error happened, the function would return `-1`.  We can check whether the return value greater equal to zero to determine the execution status.